framework/src/modules/chain/components/storage/entities/block_temp.js

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const path = require('path');
+const assert = require('assert');
+const { defaults, pick } = require('lodash');
+const {
+	entities: { BaseEntity },
+	utils: {
+		filterTypes: { NUMBER, TEXT },
+	},
+} = require('../../../../../components/storage');
+
+const sqlFiles = {
+	create: 'block_temp/create.sql',
+	delete: 'block_temp/delete.sql',
+	get: 'block_temp/get.sql',
+};
+
+/**
+ * Block temp
+ * @typedef {Object} block_temp
+ * @property {string} id
+ * @property {number} height
+ * @property {json} block
+ */
+
+/**
+ * Round Filters
+ * @typedef {Object} filters.Round
+ * @property {number} [id],
+ * @property {string} [id_eql],
+ * @property {string} [id_ne],
+ * @property {string} [id_in],
+ * @property {string} [id_like],
+ * @property {number} [height]
+ * @property {number} [height_eql]
+ * @property {number} [height_ne]
+ * @property {number} [height_gt]
+ * @property {number} [height_gte]
+ * @property {number} [height_lt]
+ * @property {number} [height_lte]
+ * @property {number} [height_in]
+ */
+
+class BlockTemp extends BaseEntity {
+	/**
+	 * Constructor
+	 * @param {BaseAdapter} adapter - Adapter to retrieve the data from
+	 * @param {filters.Round} defaultFilters - Set of default filters applied on every query
+	 */
+	constructor(adapter, defaultFilters = {}) {
+		super(adapter, defaultFilters);
+
+		this.addField('id', 'string', { filter: TEXT });
+		this.addField('height', 'number', { filter: NUMBER });
+		this.addField('fullBlock', 'string');
+
+		this.extendDefaultOptions({ sort: '' });
+
+		this.sqlDirectory = path.join(path.dirname(__filename), '../sql');
+
+		this.SQLs = this.loadSQLFiles('block_temp', sqlFiles, this.sqlDirectory);
+	}
+
+	/**
+	 * Create block_temp row entry
+	 *
+	 * @param {Object} data
+	 * @param {Object} [_options]
+	 * @param {Object} [tx] - Transaction object
+	 * @return {null}
+	 */
+	// eslint-disable-next-line no-unused-vars
+	create({ height, id, fullBlock }, _options = {}, tx = null) {
+		assert(height && Number.isInteger(height), 'height must be a number');
+		assert(id && typeof id === 'string', 'id must be a string');
+		assert(typeof fullBlock === 'object', 'block must be an object');
```

Comment:
Typeof works good only for primitives, I would prefer to use 
``
fullBlock instanceof Object
``

Reply:
Why is `_options` underscored?

Reply:
simply using the existing interface defined in the rest of the application